presto-main/src/main/java/com/facebook/presto/execution/QueryTracker.java

```diff
@@ -234,35 +237,58 @@ private void enforceTimeLimits()
         }
     }
 
+    private static class QueryAndTaskCount<T extends TrackedQuery>
```

Comment:
My understanding is that the runningTaskCount can change and hence for the sake of stability - I want to get the value and just use it for the course of this function - so that our heaps can function as expected